### ISSUE TITLE
Question about deleting a full Episode of Care (including all versions)

### ISSUE THREAD

**Issue by Remcovanderwerf** [scandihealth/lpr3-docs#200]
Hi,

i would like to know how to DELETE a complete Episode of Care from the LPR3 registry including all previously registered versions. I couldn't find it documented on the github-lpr3-docs.

For example;
an EoC has been submitted with in total 4 versions / different documents.

Now i would like to delete all these  4 versions from the registry i.e. the full EoC.

I found that i can use the template 1.2.208.176.7.1.10.116 DK Nullify Section for this, correct?
I found that in this section I need to select the template 1.2.208.176.7.1.10.65 DK External Episode Of Care Act, correct?

In this act you have to supply the setID and a document ID.
Should i supply the document ID from 1st version of the EoC or how does this work?

Thank you in advance!

Kind regards,

Remco van der Werf
Enovation | The Netherlands
Contractor: Region Syddanmark

**Comment by jonigkeit** [comment]
Hi Remco

The are two mechanisms that can be used for deleting episode of cares from the registry.

1. Create an addendum and nullify each episode of care via a nullify act
2. Create an replacement with an empty nullify section

Regardless of choice you must always send the latest document id.

Unfortunately mechanism 2 is blocked by #28 which has awaiting approval from SDS since January.

**Comment by jonigkeit** [comment]
Good news!
#28 has been approved and is planned to be included [this release](https://github.com/scandihealth/lpr3-docs/milestone/26)

**Comment by Remcovanderwerf** [comment]
Sounds great!
Then I will implement mechanism 2 for now and await testing until the next release has been deployed.

Thank you!

**Comment by metodikamikaelnilsson** [comment]
Can you please clarify some things regarding this issue.

>Create a replacement with an empty nullify section.

I did this but got this error back:

> SCHEMATRON|||(DKNullifySection): element hl7:entry[hl7:act[hl7:templateId[@root='1.2.208.176.7.1.10.118']][hl7:templateId[@root='2.16.840.1.113883.10.12.301']]] is mandatory [min 1x].

>Location: 3D57A3AB-3742-44F2-B40E-D85E97FB693D^D2704F6A-F70E-40AF-9FD4-8E3FE679CCE5|||/*:ClinicalDocument[namespace-uri()='urn:hl7-org:v3'][1]/*:component[namespace-uri()='urn:hl7-org:v3'][1]/*:structuredBody[namespace-uri()='urn:hl7-org:v3'][1]/*:component[namespace-uri()='urn:hl7-org:v3'][1]/*:section[namespace-uri()='urn:hl7-org:v3'][1]

Also, you say that the latest document ID has to be sent. Where in the clinical document should this ID be placed? Is "the latest document ID" the ID of the parent document? I've set the id@extension to the same ID as the parentDocument/id@extension, is this correct?

This is the clinical document that I sent:

```xml
<ClinicalDocument xmlns="urn:hl7-org:v3" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" classCode="DOCCLIN" moodCode="EVN">
   <realmCode code="DK" />
   <typeId extension="POCD_HD000040" root="2.16.840.1.113883.1.3" />
   <templateId root="1.2.208.176.7.1.10.71" />
   <templateId root="2.16.840.1.113883.10.12.2" />
   <templateId root="2.16.840.1.113883.10.12.1" />
   <id root="3D57A3AB-3742-44F2-B40E-D85E97FB693D" extension="B7D1C8CC-93B2-4364-A82B-45248CC23DCA" />
   <code code="34133-9" codeSystem="2.16.840.1.113883.6.1" codeSystemName="LOINC" displayName="Summary of episode note" />
   <effectiveTime value="20181129134624+0100" />
   <confidentialityCode code="N" codeSystem="2.16.840.1.113883.5.25" />
   <setId root="3D57A3AB-3742-44F2-B40E-D85E97FB693D" assigningAuthorityName="Aleris-Hamlet Hospitaler" />
   <versionNumber value="46" />
   <recordTarget nullFlavor="NA">
      <templateId root="1.2.208.176.7.1.10.101" />
      <templateId root="2.16.840.1.113883.10.12.101" />
      <patientRole classCode="PAT">
         <templateId root="1.2.208.176.7.1.10.106" />
         <id extension="0805904844" root="1.2.208.176.1.2" assigningAuthorityName="CPR" />
      </patientRole>
   </recordTarget>
   <author typeCode="AUT">
      <templateId root="1.2.208.176.7.1.10.102" />
      <templateId root="2.16.840.1.113883.10.12.102" />
      <time value="20181129134624+0100" />
      <assignedAuthor classCode="ASSIGNED">
         <id nullFlavor="NI" />
         <representedOrganization classCode="ORG" determinerCode="INSTANCE">
            <id assigningAuthorityName="SOR" root="1.2.208.176.1.1" extension="291651000016009" />
         </representedOrganization>
      </assignedAuthor>
   </author>
   <custodian typeCode="CST">
      <templateId root="1.2.208.176.7.1.10.103" />
      <templateId root="2.16.840.1.113883.10.12.104" />
      <assignedCustodian classCode="ASSIGNED">
         <representedCustodianOrganization classCode="ORG" determinerCode="INSTANCE">
            <id root="2.16.840.1.113883.2.24.1.1" extension="71017516" assigningAuthorityName="Erhvervs- og Selskabsstyrelsen" />
         </representedCustodianOrganization>
      </assignedCustodian>
   </custodian>
   <relatedDocument typeCode="RPLC">
      <parentDocument classCode="DOCCLIN" moodCode="EVN">
         <id root="3D57A3AB-3742-44F2-B40E-D85E97FB693D" extension="B7D1C8CC-93B2-4364-A82B-45248CC23DCA" />
         <setId root="3D57A3AB-3742-44F2-B40E-D85E97FB693D" />
         <versionNumber value="43" />
      </parentDocument>
   </relatedDocument>
   <component contextConductionInd="true">
      <structuredBody>
         <component contextConductionInd="true">
            <section>
               <templateId root="1.2.208.176.7.1.10.116" />
               <templateId root="2.16.840.1.113883.10.12.201" />
            </section>
         </component>
      </structuredBody>
   </component>
</ClinicalDocument>
```

**Comment by TueCN** [comment]
@metodikamikaelnilsson this is really a separate issue, but we can handle it here.

1. You need to specify what you want to nullify. See [DK Nullify Act](http://lpr-art-decor.westeurope.cloudapp.azure.com:8080/art-decor/decor-templates--lpr-?section=templates&id=1.2.208.176.7.1.10.118&effectiveDate=2018-04-18T11:55:23)

2. The meaning of parent document is explained here: https://scandihealth.github.io/lpr3-docs/aspects/index.html#documents-and-versioning

**Comment by TueCN** [comment]
1. Sorry I misunderstood your problem. The nullify section should indeed be allowed to be empty (enhancement #28). However, the Schematron rules that was updated with #28 have not actually been deployed to the service. This is an oversight and will be fixed.

2. The picture https://scandihealth.github.io/lpr3-docs/img/L-cda_figure1-changed.png illustrates what IDs are involved and where they should be placed. 
As you can see in the drawing, when replacing/appending to an existing document, the `ClinicalDocument/id` (both root and extension) of the existing document should be placed in the `ClinicalDocument/relatedDocument/parentDocument/id` of the new document. The same goes for `versionNumber`

**Comment by metodikamikaelnilsson** [comment]
2. Ok. So the ID-procedure for when deleting a full EoC by only sending an empty nullify section is exactly the same as for any other replacement. Got it.

Thank you for the clarification.

**Comment by TueCN** [comment]
1. The updated schematron rules were deployed yesterday, but the functionality in the service is not yet implemented. This means that you no loger get a schematron error, but nothing is actually deleted.

In short: We are still working on #28

**Comment by TueCN** [comment]
@metodikamikaelnilsson 
#28 has been closed and using an empty nullify section during document replacement should work as expected.

See https://github.com/scandihealth/lpr3-docs/issues/28#issuecomment-447816394
